File: tests/resources/test_codegen/test_default_code_generators_and_templates/cpp/expected_echo_generated_code.cpp
#include <iostream>
#include <vector>
#include <string>
#include <cassert>

const int mod = 123;
const std::string YES = "yes";
const std::string NO = "NO";
void solve(long long N, long long M, std::vector<std::vector<std::string>> H, std::vector<long long> A, std::vector<long double> B, long long Q, std::vector<long long> X){
    std::cout << N << " " << M << std::endl;
    assert(H.size() == N - 1);
    for (int i = 0; i < N - 1; i++) {
        assert(H[i].size() == M - 2);
        for (int j = 0; j < M - 2; j++) {
            std::cout << (j > 0 ? " " : "") << H[i][j];
        }
        std::cout << std::endl;
    }
    assert(A.size() == N - 1);
    assert(B.size() == N - 1);
    for(int i = 0;i < N - 1;i++){
        std::cout << A[i] << " " << B[i] << std::endl;
    }
    std::cout << Q << std::endl;
    assert(X.size() == M + Q);
    for(int i = 0;i < M + Q;i++){
        std::cout << X[i] << std::endl;
    }

    std::cout << YES << std::endl;
    std::cout << NO << std::endl;
    std::cout << mod << std::endl;

}

int main(){
    long long N;
    std::cin >> N;
    long long M;
    std::cin >> M;
    std::vector<std::vector<std::string>> H(N-2+1, std::vector<std::string>(M-1-2+1));
    for(int i = 0 ; i < N-2+1 ; i++){
        for(int j = 0 ; j < M-1-2+1 ; j++){
            std::cin >> H[i][j];
        }
    }
    std::vector<long long> A(N-2+1);
    std::vector<long double> B(N-2+1);
    for(int i = 0 ; i < N-2+1 ; i++){
        std::cin >> A[i];
        std::cin >> B[i];
    }
    long long Q;
    std::cin >> Q;
    std::vector<long long> X(M+Q);
    for(int i = 0 ; i < M+Q ; i++){
        std::cin >> X[i];
    }
    solve(N, M, std::move(H), std::move(A), std::move(B), Q, std::move(X));
    return 0;
}

File: tests/resources/test_codegen/test_default_code_generators_and_templates/cs/expected_echo_generated_code.cs
using System;
using System.Text;
using System.Linq;
using System.Collections;
using System.Collections.Generic;
using static System.Console;
using static System.Math;
using System.Diagnostics;

public class Program{
    const long MOD = 123;
    const string YES = "yes";
    const string NO = "NO";

    public static void Main(string[] args){
        ConsoleInput cin = new ConsoleInput(Console.In, ' ');
        long N;
        N = cin.ReadLong;
        long M;
        M = cin.ReadLong;
        string[,] H = new string[N-2+1,M-1-2+1];
        for(int i = 0;i < N-2+1;i++){
            for(int j = 0;j < M-1-2+1;j++){
                H[i,j] = cin.Read;
            }
        }
        long[] A = new long[N-2+1];
        double[] B = new double[N-2+1];
        for(int i = 0;i < N-2+1;i++){
            A[i] = cin.ReadLong;
            B[i] = cin.ReadDouble;
        }
        long Q;
        Q = cin.ReadLong;
        long[] X = new long[M+Q];
        for(int i = 0;i < M+Q;i++){
            X[i] = cin.ReadLong;
        }
        new Program().Solve(N, M, H, A, B, Q, X);
    }

    public void Solve(long N, long M, string[,] H, long[] A, double[] B, long Q, long[] X){
        WriteLine($"{N} {M}");
        Debug.Assert(H.GetLength(0) == N - 1);
        for (int i = 0;i < N - 1;i++) {
            Debug.Assert(H.GetLength(1) == M - 2);
            for (int j = 0;j < M - 2;j++) {
                Write((j > 0 ? " " : "") + $"{H[i,j]}");
            }
            WriteLine();
        }
        Debug.Assert(A.Length == N - 1);
        Debug.Assert(B.Length == N - 1);
        for(int i = 0;i < N - 1;i++){
            WriteLine($"{A[i]} {B[i]}");
        }
        WriteLine(Q);
        Debug.Assert(X.Length == M + Q);
        for(int i = 0;i < M + Q;i++){
            WriteLine(X[i]);
        }

        WriteLine(YES);
        WriteLine(NO);
        WriteLine(MOD);

    }
}

public class ConsoleInput{
    private readonly System.IO.TextReader _stream;
    private char _separator = ' ';
    private Queue<string> inputStream;
    public ConsoleInput(System.IO.TextReader stream, char separator = ' '){
        this._separator = separator;
        this._stream = stream;
        inputStream = new Queue<string>();
    }
    public string Read{
        get{
            if (inputStream.Count != 0) return inputStream.Dequeue();
            string[] tmp = _stream.ReadLine().Split(_separator);
            for (int i = 0; i < tmp.Length; ++i)
                inputStream.Enqueue(tmp[i]);
            return inputStream.Dequeue();
        }
    }
    public string ReadLine { get { return _stream.ReadLine(); } }
    public int ReadInt { get { return int.Parse(Read); } }
    public long ReadLong { get { return long.Parse(Read); } }
    public double ReadDouble { get { return double.Parse(Read); } }
    public string[] ReadStrArray(long N) { var ret = new string[N]; for (long i = 0; i < N; ++i) ret[i] = Read; return ret;}
    public int[] ReadIntArray(long N) { var ret = new int[N]; for (long i = 0; i < N; ++i) ret[i] = ReadInt; return ret;}
    public long[] ReadLongArray(long N) { var ret = new long[N]; for (long i = 0; i < N; ++i) ret[i] = ReadLong; return ret;}
}
